### ISSUE TITLE
opening book policy for multiple engines tournament

### ISSUE THREAD

**Issue by gahtan-syarif** [Disservin/fastchess#742]
in a multiple engines tournament, ideally each pair of engines uses the same openings. so engine A vs Engine B and Engine A vs Engine C should get the same opening order. this is currently not the case as the opening book index is currently incremented by rounds

**Comment by Disservin** [comment]
what is cutechess doing in this case?

**Comment by gahtan-syarif** [comment]
> what is cutechess doing in this case?

it has multiple different policys for this but i think we just need to implement the default https://github.com/cutechess/cutechess/blob/master/docs/cutechess-cli.6.txt#L197-L201

**Comment by gahtan-syarif** [comment]
I think the easiest way to do this is to make the matchmaking so that all matches between two engines are played first before it moves on to the next pairing

**Comment by Disservin** [comment]
i think you only need to refetch a new opening if the round increases ? https://github.com/Disservin/fastchess/blob/master/app/src/matchmaking/tournament/roundrobin/match_generator.hpp#L57

**Comment by gahtan-syarif** [comment]
Yeah but rn the round increases every two games so say if engine A vs engine B is round 1 then engine A vs engine C is round 2 so the opening for the 2nd pairing would be different from the first
